Automattic/WP-Job-Manager
# Add promote jobs API endpoint

Fixes https://github.com/Automattic/WP-Job-Manager/issues/2437

### Changes proposed in this Pull Request

* Add a REST API endpoint to show all promoted jobs at `/wp-json/wpjm-internal/v1/promoted-jobs`

### Testing instructions

* Check out this PR
* Make a few job listings promoted by adding the `_promoted => 1` post meta
* Go to the REST API endpoint to ensure they show up there.  Example:`http://wpjm.local/wp-json/wpjm-internal/v1/promoted-jobs`

### Screenshot / Video
<img width="925" alt="Screenshot 2023-06-21 at 3 27 19 PM" src="https://github.com/Automattic/WP-Job-Manager/assets/3220162/6a5bf0ee-2932-4039-880f-dd31840e554a">
